src/InfiniteIterableValue.php

```diff
@@ -246,9 +268,39 @@ static function (iterable $iterable) use ($offset, $length): iterable {
         ));
     }
 
+    /**
+     * @phpstan-return InfiniteIterableValue<TKey, TValue>
+     */
+    public function skip(int $length): InfiniteIterableValue
+    {
+        return self::fromStack($this->stack->push(
+            /**
+             * @phpstan-param iterable<TKey, TValue> $iterable
+             * @phpstan-return iterable<TKey, TValue>
+             */
+            static function (iterable $iterable) use ($length): iterable {
+                foreach ($iterable as $key => $value) {
+                    if ($length-- > 0) {
+                        continue;
+                    }
+
+                    yield $key => $value;
+                }
+            }
+        ));
```

Comment:
Not sure what I would expect when input `$length <=0`:
a) throw error
b) skip `$length` elements from the end (not quite possible for iterator) 
c) just return `$this`
:thinking:

Reply:
`nikic/iter` goes with a) https://github.com/nikic/iter/blob/master/src/iter.php#L509